### DISCUSSION TITLE
Fan installation and control via TP4

### DISCUSSION THREAD

**Discussion by anhpham140195** [theyosh/TerrariumPI#860] [Q&A] [answered]
Hello, 

I'm a newbie, I am looking for a way to be able to install 1-2 fans and be able to control it via TP4, as I understand it can be done via Relay

I did a lot of research on the internet but it mostly just confused me :( 

Please help me, with which devices should I buy and how to connect them to RPi, mine is RPi zero 2 W.

Currently, I adr have 2 fans with 5v.

Thanks a lot!

**Comment by InsertAnyName** [comment]
I used these relays and they work: https://www.ebay.de/itm/155086624324, but you may also use a multichannel relay.

Add your relay to TP
Hardware: Relays are always GPIO, if your relay switches when there is no power, select GPIO Inverse.
Address: Address the pin number
Name: As you want
Power usage: Power of your consumer, you can also set it to 0, but it is interesting to know how much the system uses
 Timer: disable if you want to control the relay with a sensor alarm, enable as a time switch
Timer:
             For 24-hour operation, leave the timer start and stop at 00:00.
             Timer on duration: Time in which the consumer is running
             Timer off duration: Time during which the consumer is off
                 Example: on:10, off:20 The consumer runs for 10 minutes and is then off for 20 minutes

Sometimes a reboot is needed to make sure changes are used by TP

I used jumper cables to connect the following pins of the relay to the Raspberry Pi:
VCC -> Pin 2/4 5v Power
GND -> Pin 6, 14 or 20 Ground
IN/S -> Pin 16, 18, 22 etc. GPIO

Each channel of the relay board has 3 screw terminals, in the middle is usually the input, the right and left one are either open or close in idle state. (use a continuity tester to test). Attach unswitched 5V + to the middle and attach the output channel to your fan.

**Comment by theyosh** [comment]
Maybe this will help a bit: https://raspberrypi-guide.github.io/electronics/control-electronics-with-a-relay#what-is-a-relay

Is you fan a 220 volt or 12 volt? If it is just 12 volt, it is a bit easier and safer :)

**Reply by anhpham140195** [reply]
@theyosh my fan is 12v.

After checking carefully I want to control the fan 12v with IRF520 relay and power from a 9v battery: 
[9v battery image](https://vppminhthy.com/uploads/source/energizer-9v-1.jpg)

Is it ok? 

And it is not possible to use the power directly from RPi to run the fan, right?

Thank you!

**Comment by InsertAnyName** [comment]
The GPIO pins of the Raspberry Pi deliver only 3.3 V and 16mA each / 51mA in total over all GPIO. That might not be enough even for the 5V fan you mentioned in your initial post. Is there a reason to use a battery? You need a power supply for the Pi anyways.
The IRF520 Switching-Modul looks fine to me. Be careful to select one that can handle the current you need. There are some available that can handle only 1A if not cooled.

**Reply by anhpham140195** [reply]
thanks so much @InsertAnyName

**Reply by anhpham140195** [reply]
HI @InsertAnyName, about the reason use a battery, I did not see this yesterday. 
I mean the external battery for the fan 12v.
In other cases if I use fan 5v can I just directly plug in the GPIO pins of the RPi with 5v pin, is it possible? Thanks

**Comment by InsertAnyName** [comment, marked as answer]
I do not think you can use the 5V fan directly at the GPIO pins. The fan with the lowest current I found during a quick search was 0,2A. I would suggest to use a power supply (maybe you have one from an old telephone or LED light) check the V and W it can deliver, connect it to the IRF520  and run your fan with it. You may also buy an new power supply. They cost approx. the price of 2-3 9V blocks. Don't be scared about the wiring. It is simple.
